src/main/java/mat/server/service/jobs/CheckUserLastLoginTask.java

```diff
@@ -210,7 +210,8 @@ private void checkUserLoginDays(final long noOfDayLimit, final String emailType)
 
             lastSignInDate = DateUtils.truncate(lastSignInDate, Calendar.DATE);
             logger.info(USER_LOG_LABEL + user.getFirstName() + "  :::: lastSignInDate :::::   " + lastSignInDate);
-            if (lastSignInDate.equals(daysAgo)) {
+            if (emailType.equals(WARNING_EMAIL_FLAG) && lastSignInDate.equals(daysAgo)

```

Comment:
You could refactor this into a private method that takes an emailType dateToCheck emailType and daysAgo as params and remove the duplicate code here and in 201/202 would prob make the if statement easier to read as well.

Reply:
Nice idea!